### PR TITLE
feat(equivalence): a cross-axis rung, and an undecidable one

### PR DESCRIPTION
**Pull request by nh13**
## The circularity, and the way out

`normalize` maps description → description. Its job is to find one canonical output for every input in an equivalence class, so a definition of "equivalent" may not itself call `normalize` — "a and b are equivalent when they normalize alike" is true of every deterministic function and measures nothing. `EquivalenceChecker`'s `NormalizedMatch` rung is exactly that definition, it is the rung most reachable in practice, and a gate written over `is_equivalent()` silently accepts it.

`apply` escapes it. `apply` maps description → **sequence**, and equality on the codomain is byte equality on bases — not a function of ferro's normalizer, of the input's spelling, or of any choice this project makes. So the relation is stated over `apply` and the gate over `normalize`, and the two cannot collapse into each other.

## Single-axis apply-equality is not enough

`NM_004006.2:c.3921dup` and `NM_004006.2:c.3922dup` are apply-equal **on the transcript** — both positions carry `T`, so applying either yields one string. They denote different things on the genome: the positions sit on opposite sides of an exon junction and project ~2,790 bp apart, into different exons. `DNA/duplication.md:148` says so in as many words — one of the two lands on "the wrong nucleotide, in the wrong exon".

So apply-equality on the axis a description happens to be *written in* does not establish that two descriptions denote one variant. The relation has to quantify over every axis the description determines.

## Four changes to `src/equivalence/checker.rs`

1. **`CrossAxisSequenceMatch`** — apply-equal on every *determined* axis, not just the description's own. An axis is determined when the description's coordinates can be carried to it by a mapping that is a function of **reference data alone** — exon alignments, CDS offsets — and never of normalization. A `c.`/`n.`/`r.` description determines two (the transcript, and the genome its exon alignment carries it to); a `g.`/`m.` description determines only the genome. Routed through `CoordinateMapper::tx_to_genomic`, which holds nothing but a `&Transcript` and does exon arithmetic — verified normalize-free, which is the load-bearing property of the whole change. The one whole-variant API that would reintroduce the circularity, `VariantProjector::project_to_genomic_normalized`, is deliberately not used.

2. **`SequenceMatch`'s doc corrected** — it is own-axis only: true, and *insufficient for identity*, with the `c.3921dup` counterexample inline so the limit is read at the point the rung is read.

3. **`Indeterminate` + `is_decided()`** — `NotEquivalent` used to mean both "these denote different sequences" and "no denotation could be computed", and `is_equivalent()` answered false to both, so an undecidable pair was published as a decided negative. `is_decided()` is the predicate a census reads; it is false only for `Indeterminate`.

4. **Rung ordering** via `is_at_least()`, with `NormalizedMatch` deliberately **off the order in both directions** so the circular gate cannot be written by accident.

Python bindings mirrored. The new discriminants are **appended** (5/6), so the ABI is preserved.

## Tests

`tests/it/equivalence_cross_axis_rung.rs`, 14 tests, including the counterexample built at the real **minus-strand** `NM_004006.2` geometry — which matters because minus strand is the other arm of `tx_to_genomic` and the arm a plus-strand fixture would never exercise.

## Relationship to #1609

#1609 adds `src/equivalence/key.rs` with `spdi_key`/`is_confluent`. The two are kept distinct and cross-referenced from `src/equivalence/mod.rs`; they are not redundant, and they demonstrably disagree.

A pairwise verdict is not a bucketing key and a key is not a pairwise verdict. `spdi_key` reads a canonical SPDI off the resulting bases on the description's **own** accession, so it is the single-axis relation promoted to something `Hash + Ord`. The disagreement is the axis count, and this branch's counterexample is the witness: `spdi_key` buckets `c.3921dup` **together with** `c.3922dup`, while `CrossAxisSequenceMatch` separates them.

## Open questions this branch states rather than papers over

These are recorded here because each is a real residue, not a loose end a reviewer should assume is handled:

- **Where `AccessionVersionDifference` sits.** Implemented as off the order, on the grounds that apply-equality is not defined across two different reference sequences. That is sound, but it is a **house choice, not spec-derived** — no clause ranks it.
- **When the *second* axis cannot be compared** (`WindowTooWide`, `Declined`, or no genomic mapping), the pair reports `SequenceMatch`, which `is_decided()` calls true. Reviewed and kept: own-axis equality *was measured* and agreed, so reporting `Indeterminate` would discard a fact rather than withhold one, and the gate is unaffected because it reads the level — `is_at_least(CrossAxisSequenceMatch)` rejects `SequenceMatch` either way. What the level still does not carry is the difference between "cross-axis answered no" and "cross-axis could not be asked"; that survives in the note. The choice now lives on the arm in `strengthen_across_axes` rather than only here, and `a_second_axis_that_cannot_be_computed_stops_at_the_single_axis_rung` pins it with a control that reaches the cross-axis rung on the same pair.
- **Rings still return `Err(UnsupportedVariant)`** rather than `Indeterminate`. Deliberate: that error is the contract pinned by `issue_1578_followup_equivalence_rungs`, and moving it would be an unrelated behaviour change.

## Re-blessing the clause index

The rebase onto current `main` left `tests/it/clause_ruling_index.rs` carrying `main`'s pins, which are stale with respect to the ruling record this branch adds. The second commit regenerates the embedded index block from the ledger — the block the module's own `the_rendered_index_is_current` prints — and moves the two census pins to the measured values: records/decided/undecided `(20, 17, 3)` → `(21, 18, 3)`, and multiply-named clause lines `17` → `19`. The mixed-verdict count is unchanged at 7 — it reached 7 on `main`, via `delins-recommendation-reach-when-the-input-arrives-split`, and this branch's record adds no spread to any clause line. Every number is one the suite measured.

## Rebased onto `ce933533`

Two conflicts, both in committed files. `hgvs_spec_normalization_overrides.json` was add/add — `main`'s `projection-codon-exception-is-decided-by-the-rendered-axis` against this branch's record — and both are kept; a set comparison against `origin/main` and against the pre-rebase tip confirms no record was dropped on either side and no status changed. `reported_partition_verdicts.rs` was semantic: `main` had derived the expected level per pair (`NormalizedMatch` where the two spellings print one string, after #1649), while this branch asserted one rung for every pair. The resolution keeps `main`'s per-pair derivation and supplies this branch's rung as the non-converged case.

**#1649 also invalidated two of this branch's own tests, and the way it did so is the reason for a new helper.** They were written over #1419's partition pair on the ground that it stays in two canonical forms; #1649 converged it, so both tests began asserting against `NormalizedMatch` — passing a level check while exercising no denotational rung at all. They now use #1420-v2's `dup`+`del` partition, which the splitter does not reach, and `assert_normalized` pins the pair of normalized strings every rung is reached over, so a repeat of that convergence fails by name instead of silently.

`a_span_past_the_compare_window_cap_is_indeterminate` moved off a 200 kb homopolymer onto an `ACGT` repeat. On the homopolymer both deletions 3'-shuffle to the normalizer's fetch edge, so the verdict was a function of `NormalizeConfig::window_size` rather than of the cap the test names — not hypothetical, since #1697 grew that window and this test is what surfaced the resulting non-idempotent normalization. On `ACGT` repeats neither deletion can move at any window size, so the union span is the only free variable; a near-pair control pins that the cap is what decides it.

Representation-Change: none. No watched directory is touched — the change is confined to `src/equivalence/`, the bindings and tests.

## Summary by CodeRabbit

* **New Features**
  * Added cross-axis sequence matching for stronger transcript and genomic equivalence results.
  * Added an indeterminate result for comparisons that cannot be conclusively decided.
  * Added checks for equivalence decision status and minimum equivalence levels.
  * Improved strand-aware transcript-to-genomic comparison and handling of unsupported variants.

* **Documentation**
  * Clarified equivalence levels, ordering, cross-axis validation, and normalization requirements.

* **Tests**
  * Expanded coverage for cross-axis matching, indeterminate results, equivalence thresholds, and edge cases.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai pause

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

File: src/equivalence/mod.rs [hunk 1]
```diff
@@ -21,17 +21,51 @@
...
+//!   (`DNA/duplication.md:148`)
 //! - **AccessionVersionDifference**: Same variant, different accession versions
 //! - **NotEquivalent**: Represent different changes
+//! - **Indeterminate**: Neither — at least one side has no computable
+//!   denotation. Not a negative verdict; see [`EquivalenceLevel::is_decided`]
+//!
+//! [`EquivalenceLevel::is_at_least`] expresses the order over the denotational
+//! rungs, and deliberately makes **NormalizedMatch unusable as a gate**: that
+//! rung consults the normalizer, so gating on it would restore the circularity
+//! above.
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/equivalence/mod.rs:68]: Kept as `SequenceMatch`, deliberately — with the reasoning moved out of the PR description and onto the code, plus a test, so it stops being a claim nobody can check.

`Indeterminate` means *no denotation could be computed for a side*. On these paths both sides denote, the own-axis comparison ran, and it **agreed**; what is missing is corroboration from the second axis, not a denotation. Returning `Indeterminate` would discard a measured fact rather than withhold an unmeasured one, and would answer "cannot tell" about a pair whose own-axis equality is established.

Nothing is over-claimed by keeping it decided, because a gate reads the **level**: a confluence gate is written `is_at_least(CrossAxisSequenceMatch)`, and `SequenceMatch` fails that in both readings. The pair is correctly excluded either way; the only difference is whether the true own-axis fact survives in the answer.

The half of this that was right is that the residue was recorded only in prose. That is now fixed:

- the choice and its grounds sit on the `WindowTooWide | Declined` arm in `strengthen_across_axes`;
- `a_second_axis_that_cannot_be_computed_stops_at_the_single_axis_rung` pins the `NotComputed` path (a transcript served without a contig, so the own-axis comparison still succeeds), asserts the verdict is decided and does **not** clear a cross-axis gate, asserts the note records *why*, and carries a control showing the same pair reaches `CrossAxisSequenceMatch` once the contig is restored;
- the `Indeterminate` doc no longer implies the compare-window cap always lands there — it now says own-axis, and points at this arm for the second-axis case.

What the level still does not carry is "the second axis disagreed" versus "it could not be asked". That distinction lives in the note. Splitting it into the level needs a rung of its own rather than a reuse of `Indeterminate`, and that is worth doing when a consumer is measured needing it.
